buildkite/pipelines/bacalhau-golang.yaml

```diff
@@ -0,0 +1,64 @@
+cache:
+  paths:
+    - "/golangci-lint-cache"
+    - "/vendor-cache"
+    - "/node-modules-cache"
+  size: 20g
+  name: "vendor-module-cache"
```

Comment:
how is this cache used? We also want to make sure we build once, and then use the builds in later phases, such as running unit tests, integration tests, and releasing them if this build includes a release

Reply:
also do we pull the code once, or every step in the pipeline pulls the code again?

Reply:
Every step in pipeline unfortunately pulls code :( But we use git mirroring which fastens up. 

If you want to do just once. You cant take advantage of parallel builds. 

Circle CI was the same too.